.github/workflows/build_and_test.yaml

```diff
@@ -16,31 +16,51 @@ on:
       - "**/*.png"
 env:
   GO_VERSION: 1.18.2
+  ENVOY_GATEWAY_DEV_IMAGE: envoyproxy/gateway-dev
+  ENVOY_GATEWAY_DEV_TAG: latest
 jobs:
-  build:
+  build-and-test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
           go-version: ${{ env.GO_VERSION }}
           cache: true
-      - name: build
-        run: make build
-  test:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
-        with:
-          go-version: ${{ env.GO_VERSION }}
-          cache: true
-      - name: test and report coverage
+      - name: Test and report coverage
         run: go test ./... -race -coverprofile=coverage.xml -covermode=atomic
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v2
         with:
           fail_ci_if_error: true
           files: ./coverage.xml
           name: codecov-envoy-gateway
-          verbose: true
\ No newline at end of file
+          verbose: true
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          platforms: linux/amd64
+          tags: ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ github.sha }}
```

Comment:
do we want both the commit SHA and `latest` for tags here?

Reply:
`latest` maps to push events on `main`, so prefer not to add it here since these steps which apply to all events (`push` & `pr_request`)